tiledb/sm/c_api/tiledb.cc

```diff
@@ -5913,23 +5912,22 @@ int32_t tiledb_deserialize_array(
   }
 
   // Allocate an array object
-  (*array)->array_ =
-      new (std::nothrow) tiledb::sm::Array(uri, ctx->ctx_->storage_manager());
-  if ((*array)->array_ == nullptr) {
-    delete *array;
-    *array = nullptr;
+  try {
+    (*array)->array_ = make_shared<tiledb::sm::Array>(
+        HERE(), uri, ctx->ctx_->storage_manager());
+  } catch (...) {
```

Comment:
Should we be catching std::bad_alloc here too?

Reply:
Forgot to change, fixing now